scio-core/src/main/java/com/spotify/scio/util/RemoteFileUtil.java

```diff
@@ -104,7 +106,12 @@ public Path download(URI src) {
    */
   public List<Path> download(List<URI> srcs) {
     try {
-      return paths.getAll(srcs).values().asList();
```

Comment:
`getAll(srcs)` returns a `Map<URI, Path>`. Using `values` give no guarantee that returned `Paths` are ordered.